raspberry/bzzz/sensors/gps.py

```diff
@@ -0,0 +1,220 @@
+        self.__thread.daemon = True  # Ensure thread does not prevent program termination
+        self.__thread.start()
+
+    def __get_measurements_in_background_t(self, serial_path, baud):
+        """
+        Continuously reads GPS data from the serial port, parsing GNGLL messages
+        for latitude and longitude, and GPGSV messages for altitude, 
+        updating the object's GPS attributes accordingly. This method will write
+        the GPS data to a CSV file for 2 minutes.
+
+        :param serial_path: Serial port path
+        :param baud: Baud rate for the serial connection
+        """
+        stop_time = datetime.now() + timedelta(minutes=2)  # Determine when to stop collecting data
+        with serial.Serial(serial_path, baud, timeout=1) as ser, open("gps_data.csv", "w", newline='') as file:
+            writer = csv.writer(file)
+            writer.writerow(["Timestamp", "Latitude", "Longitude", "Altitude"])  # Header row for CSV
+
+            while datetime.now() < stop_time:  # Collect and write data for 2 minutes
+                if ser.in_waiting > 0:
+                    line = ser.readline().decode('utf-8', 'ignore').strip() # use ignore when the code come accross undecodable bytes
+                    tokens = line.split(",")
+                    msg_key = tokens[0]
+
+                    if msg_key == "$GNGLL" and tokens[1] and tokens[3]:
+                        lat_deg = int(float(tokens[1]) / 100)
+                        lat_min = float(tokens[1]) % 100
+                        latitude = deg_min_sec_to_decimal(lat_deg, lat_min, tokens[2])
+
+                        lon_deg = int(float(tokens[3]) / 100)
+                        lon_min = float(tokens[3]) % 100
+                        longitude = deg_min_sec_to_decimal(lon_deg, lon_min, tokens[4])
+
+                        self.__gpgll_latitude = latitude
+                        self.__gpgll_longitude = longitude
+
+                    elif msg_key == "$GPGSV" and len(tokens) > 5:
+                        altitude = float(tokens[5])
+                        self.__gpgsv_altitude = altitude
+
+                    # Only write to CSV if all data is present
+                    if self.__gpgll_latitude is not None and self.__gpgll_longitude is not None and self.__gpgsv_altitude is not None:
+                        writer.writerow([datetime.now().strftime("%Y-%m-%d %H:%M:%S"), self.__gpgll_latitude, self.__gpgll_longitude, self.__gpgsv_altitude])
+
+    @property
+    def latitude(self):
+        """
+        Latitude in decimal degrees
+        """
+        return self.__gpgll_latitude
+
+    @property
+    def longitude(self):
+        """
+        Longitude in decimal degrees
+        """
+        return self.__gpgll_longitude
+
+    @property
+    def altitude(self):
+        """
+        Altitude in meters
+        """
+        return self.__gpgsv_altitude
+
+if __name__ == "__main__":
+    gps = Gps()  # Using default serial path and baud rate
```

Comment:
Use the syntax 

```python
with Gps() as gps:
    time.sleep(120)
```

For this to work you need to implement `__enter__` and `__exit__`. Can you see how we did this for the [anemometer](https://github.com/QUB-ASL/bzzz/blob/dev/raspberry/bzzz/sensors/anemometer.py) and do the same here?